src/software/ai/hl/stp/tactic/shadow_freekicker_tactic_test.cpp

```diff
@@ -9,86 +9,858 @@
+ *            |         friendly    friendly        |
+ *            |                                     |
+ *            |                                     |
+ *            |                                     |
+ *            |                                     |
+ *            |                                     |
+ *            |           +-------------+           |
+ *            |           |             |           |
+ *            |           |             |           |
+ *            |           |             |           |
+ *            +-----------+-------------+-----------+
+ */
+
+TEST_F(ShadowFreekickerTacticTest,
+       test_shadow_free_kicker_left_side_without_robot_in_enemy_team)
+{
+    Point ball_position(-0.09, world.field().friendlyCornerPos().y() - 0.09);
+    ShadowFreekickerTactic::FreekickShadower left_or_right = ShadowFreekickerTactic::LEFT;
+
+    CreateSceneWithoutEnemyRobot(ball_position, left_or_right);
+
+    // The edge of the robot should be slightly more than 0.5m from the ball
+    EXPECT_NEAR(ball_to_friendly_robot_distance, EXPECTED_BALL_TO_FRIENDLY_ROBOT_DISTANCE,
+                BALL_TO_FRIENDLY_ROBOT_DISTANCE_ABS_ERROR);
+
+    // The robot should be just to the left of the line between the friendly net and
+    // the ball (from the POV of the friendly net)
+    EXPECT_NEAR(middle_line_to_friendly_robot_distance,
+                EXPECTED_MIDDLE_LINE_TO_FRIENDLY_ROBOT_DISTANCE,
+                MIDDLE_LINE_TO_FRIENDLY_ROBOT_DISTANCE_ABS_ERROR);
+
+    EXPECT_GT(goal_to_dest_angle, goal_to_ball_angle);
+}
+
+TEST_F(ShadowFreekickerTacticTest,
+       test_shadow_free_kicker_right_side_without_robot_in_enemy_team)
+{
+    Point ball_position(-0.09, world.field().friendlyCornerPos().y() - 0.09);
+    ShadowFreekickerTactic::FreekickShadower left_or_right =
+        ShadowFreekickerTactic::RIGHT;
+
+    CreateSceneWithoutEnemyRobot(ball_position, left_or_right);
+
+    // The edge of the robot should be slightly more than 0.5m from the ball
+    EXPECT_NEAR(ball_to_friendly_robot_distance, EXPECTED_BALL_TO_FRIENDLY_ROBOT_DISTANCE,
+                BALL_TO_FRIENDLY_ROBOT_DISTANCE_ABS_ERROR);
 
     // The robot should be just to the right of the line between the friendly net and
     // the ball (from the POV of the friendly net)
-    Line ball_to_net_line = Line(world.ball().position(), world.field().friendlyGoal());
-    EXPECT_NEAR(distance(ball_to_net_line, move_action->getDestination()), 0.09, 0.01);
-    Angle goal_to_ball_angle =
-        (world.ball().position() - world.field().friendlyGoal()).orientation();
-    Angle goal_to_dest_angle =
-        (move_action->getDestination() - world.field().friendlyGoal()).orientation();
+    EXPECT_NEAR(middle_line_to_friendly_robot_distance,
+                EXPECTED_MIDDLE_LINE_TO_FRIENDLY_ROBOT_DISTANCE,
+                MIDDLE_LINE_TO_FRIENDLY_ROBOT_DISTANCE_ABS_ERROR);
+
     EXPECT_LT(goal_to_dest_angle, goal_to_ball_angle);
 }
```

Comment:
Files should have EOF extra line.

Reply:
I do have an extra line in CLion for some reason, but I added another line and now github can show the EOF line